### ISSUE TITLE
Hierarchical display nodes

### ISSUE THREAD

**Issue by TonyStark-git** [vasturiano/3d-force-graph#601]
I'm trying to make a graph that is displayed hierarchically by node type,
Any way or suggestion? Thanks
similar

node type1: ----------------
                      \ | | | \ \ \XXX\\|           <- link
node type2: ----------------
                       \XXX\\| \ | | | \ \          <- link
node type3: ----------------

**Comment by TonyStark-git** [comment]
> Something like [this example](https://vasturiano.github.io/3d-force-graph/example/tree/) [(source)](https://github.com/vasturiano/3d-force-graph/blob/master/example/tree/index.html)?

Yes, but my data is not regular tree data, there are some circular relationships such as: 1-2-3-1 or 1-2-1, which caused an error in dagMode

**Comment by vasturiano** [comment]
If your data has loops like that it cannot be fully represented as a hierarchical DAG.

Having said that, you can use `onDagError` to choose to ignore the error, knowing that it will produce layouts that are not strictly hierarchical.
